tool/ood-gen/lib/exercise.ml

```diff
@@ -73,7 +73,22 @@ let decode (fpath, (head, body)) : (t, [> `Msg of string ]) result =
   in
   Ok (metadata |> of_metadata ~statement ~solution)
 
-let all () = Utils.map_files decode "exercises/*.md"
+let compare_by_slug =
+  let parse_int s = s |> int_of_string_opt |> Option.value ~default:0 in
+  let key exercise : int * string =
+    let slug = exercise.slug in
+    let len = String.length slug in
+    if len = 0 then (0, "")
+    else
+      match slug.[len - 1] with
+      | 'A' .. 'Z' as c ->
+          (parse_int (String.sub slug 0 (len - 1)), String.make 1 c)
+      | _ -> (parse_int slug, "")
```

Comment:
```suggestion
    Scanf.sscanf exercise.slug "%d%[A-Z]" (fun s c -> (s, c))
```

What do you think of something like that?

Reply:
Wow, I didn't know sscanf was so versatile -- thanks for showing me. I'll change this and remove parse_int.
This could throw at build time if sscanf fails, but I don't imagine the slug format will change drastically in the future. We can always just update the format string + receiver function if so.